leadership/eng-lead/README.md

```diff
@@ -0,0 +1,34 @@
+# [Leadership](../README.md) / Engineering Lead
+
+## Overview
+
+Guidance and resources for Application Engineering Leads.
+
+* Manager and peer support
+  * Lean on your manager and set up 1:1s with other past/present Engineering Leads for feedback or advice.
+  * Begin to ascertain any aspects of the projects you may not be as familiar with. For example, if you are concerned about any gaps in infrastructure knowledge, please reach out to someone in #infra.
+* Review the [Technical Requirements Checklist 🔒](https://docs.google.com/document/d/1ozaXs8FiI6tqRWS_mEmEx_PothTioJsOxHoYlJF0b3I/)
```

Comment:
Were did this one come from? I have questions. Not a blocker for this PR, but I would prefer to add a link to it later once the checklist has been reviewed.

Reply:
This came from @macrael , and this list helped to inspire this background work and PR since I find it helpful as a guide when starting new projects. I'm open to commenting it out for now before a review, but I would be keen on having this returned to the guide as quickly as we can since it's shown to be useful to a few folks in the practice.